--- a/src/shared/cgroup-setup.c
+++ b/src/shared/cgroup-setup.c
@@ -330,10 +330,14 @@ int cg_create(const char *controller, const char *path) {
 
         r = mkdir_parents(fs, 0755);
         if (r < 0)
-                return r;
+                if (!IN_SET(r, -EROFS, -EPERM, -EACCES, -ENOENT))
+                        return r;
 
         r = RET_NERRNO(mkdir(fs, 0755));
-        if (r == -EEXIST)
+        if (IN_SET(r, -EROFS, -EPERM, -EACCES, -ENOENT))
+                return 0;
+
+        if (r == -EEXIST)
                 return 0;
         if (r < 0)
                 return r;
@@ -370,6 +374,8 @@ int cg_attach(const char *controller, const char *path, pid_t pid) {
         r = write_string_file(fs, c, WRITE_STRING_FILE_DISABLE_BUFFER);
         if (r == -EOPNOTSUPP && cg_is_threaded(path) > 0)
                 return -EUCLEAN;
+        if (IN_SET(r, -EROFS, -EPERM, -EACCES, -ENOENT))
+                return 0;
         if (r < 0)
                 return r;
